prod/pillaralgos/helpers/emoji_getter.py

```diff
@@ -0,0 +1,159 @@
+"""
+Contains one class `emoticonExtractor` that returns a df of top emoticons used,
+including the video's ID, and the emoticon's id, name, number of uses. Plus a blank
+"label" column to be filled out by little label-monkeys.
+
+"Top emoticons used" is defined as `emoticon count in dataset` >= `mean emoticon count in dataset`
+
+HOW TO:
+    ee = emoticonExtractor(data, limit='mean')
+    vid_emos_df = ee.run()
+"""
+from pillaralgos.helpers import data_handler as dh
+
+import pandas as pd
+import json
+from pillaralgos.helpers import data_handler as dh
+
+
+class emoticonExtractor:
+    def __init__(self, data, min_use="mean", limit=None, save_csv=False):
+        """
+        Gets data ready for emo extraction. Initializes dicts, lists, etc.
+
+        input
+        -----
+        data: list
+            List of dictionaries, a json file opened with json.load(open(file))
+        min_use: str, int, None
+            'mean': Return only those emoticons who's count is > the mean occurrance
+            int: Return only those emoticons who's count is > X. Use 0 to not filter.
+        limit: int, None
+            int: Return only the top X emoticons (using df.head(X))
+            None: Return all emoticons
+        """
+
+        big_df = dh.organize_twitch_chat(data)
+        print(big_df.iloc[-1, 0] - big_df.iloc[0, 0])
+        self.vid_id = data[0]["content_id"]
+        self.big_df = big_df
+        self.all_emos = (
+            []
+        )  # all unique emo_ids, later used to pd.Series(self.all_emos).value_counts()
+        self.limit = limit
+        self.min_use = min_use
+        self.save_csv = save_csv
+
+    def run(self):
+        """
+        Coordinates all functions to return a dataset of top emojis used
+        """
+        import pandas as pd
+
+        body_has_emo = self.big_df[~self.big_df["emoticons"].isna()].reset_index(
+            drop=True
+        )
+
+        body_has_emo["emo_id_list"] = body_has_emo["emoticons"].apply(
+            lambda my_list: self.emo_extractor(my_list)
+        )
+
+        body_has_emo["emo_id_list"].apply(lambda my_list: self.emo_saver(my_list))
+        body_has_emo["emo_loc"] = body_has_emo["emoticons"].apply(
+            lambda my_list: self.loc_extractor(my_list)
+        )
+
+        emo_data = body_has_emo[["emo_id_list", "emo_loc", "body"]]
+        emo_dict = self.emo_id_matcher(
+            emo_data
+        )  # create a dictionary of emo_id:emo_name
+
+        num_used = self.number_times_used()
+        num_used["emoji_name"] = num_used["emoji_id"].map(emo_dict)
+        num_used["label"] = ""
+
+        top_emoticons = self.finalize(num_used)
+        if self.save_csv:
+            top_emoticons.to_csv(f"top_emos_strim{self.vid_id}.csv", index=False)
+        return top_emoticons
+
+    ### ACTUAL FUNCTIONS ###
+    def number_times_used(self):
```

Comment:
https://docs.python.org/3/tutorial/classes.html#private-variables use private methods for things that are only used by the class and aren't needed by the person importing the class. All this means is that you should prefix those classes with `__`

So for example 

`def __number_times_used(self):`